server/jetstream_api.go

```diff
@@ -262,6 +266,15 @@ type JSApiStreamInfoResponse struct {
 
 const JSApiStreamInfoResponseType = "io.nats.jetstream.api.v1.stream_info_response"
 
+// JSApiStreamLookupResponse.
+type JSApiStreamLookupResponse struct {
+	ApiResponse
+	Stream   string `json:"stream"`
+	Filtered bool   `json:"is_filtered"`
```

Comment:
If we go on the assumption that we'll later lookup by other properties, then this key is probably too generic

Reply:
What key? This is specific to allow developers to potentially only deal with subjects and not streams and consumers (although that should be possible).

We need to know if the subject was filtered (a subset of the streams subjects). Maybe call it Subset instead?

And reason I only return name and force another call for possibly creating consumer was to not sidestep subject based authorization for stream info etc.

Reply:
This is specific lookup by subject only. I had that in the signature but we already have way too many function and variable names that remind me of a java code base.